openshift-container-platform-4/policies/AU-Audit_and_Accountability/component.yaml

```diff
@@ -88,32 +88,35 @@
 - control_key: AU-3
   standard_key: NIST-800-53
   covered_by: []
-  implementation_status: planned
+  implementation_status: complete
   narrative:
     - text: |
-        A control response is planned. Engineering progress can be tracked via:
-
-        https://issues.redhat.com/browse/CMP-156
+        When the OpenShift API audit system is enabled and configured it generates
+        audit records containing information that establishes the type of event that occurred,
+        when the event occurred, where the event occurred, the source of hte event,
+        the outcome of the event, and the idenity of the individuales or subjects associated
+        with the event.
```

Comment:
While the statement is correct, could a sample audit log be given and each field (eg type, source, time/date) be dissected? This will help fend off auditors who want proof that OCP logs contain such information.

Reply:
We'll need to document known deficiencies too. For example, usage of ``oc debug`` does not create audit events with this information, and its use is non-attributable.

Reply:
Side note - per https://docs.openshift.com/container-platform/4.3/logging/cluster-logging.html#cluster-logging-about-fluentd_cluster-logging - 

``
The container runtimes provide minimal information to identify the source of log messages: project, pod name, and container id. This is not sufficient to uniquely identify the source of the logs. If a pod with a given name and project is deleted before the log collector begins processing its logs, information from the API server, such as labels and annotations, might not be available. There might not be a way to distinguish the log messages from a similarly named pod and project or trace the logs to their source. This limitation means log collection and normalization is considered best effort.
``

Reply:
@shawndwells I added additional content that should address the three comments above. One reference indicates the field detail fluentd sends to elastic.  If you think a specific example of a /var/log/container file or systemd journal I can add those.